### DISCUSSION TITLE
Error: Optional object type attributes are experimental

### DISCUSSION THREAD

**Discussion by jonasbenavides82** [chgasparoto/curso-aws-com-terraform#156] [Q&A]
Cleber estou tomando o erro ao dar terraform init para o site-estático, pode me ajudar com isso?

╰─➤  terraform init -backend=true -backend-config="backend.hcl"
Initializing modules...
There are some problems with the configuration, described below.

The Terraform configuration must be valid before initialization so that
Terraform can determine which modules and providers need to be installed.

Error: Optional object type attributes are experimental

  on .terraform/modules/logs/variables.tf line 74:
  74: variable "versioning" {

This feature is currently an opt-in experiment, subject to change in future
releases based on feedback.

Activate the feature for this module by adding module_variable_optional_attrs
to the list of active experiments.

Error: Optional object type attributes are experimental

  on .terraform/modules/redirect/variables.tf line 74:
  74: variable "versioning" {

This feature is currently an opt-in experiment, subject to change in future
releases based on feedback.

Activate the feature for this module by adding module_variable_optional_attrs
to the list of active experiments.

Error: Optional object type attributes are experimental

  on .terraform/modules/website/variables.tf line 74:
  74: variable "versioning" {

This feature is currently an opt-in experiment, subject to change in future
releases based on feedback.

Activate the feature for this module by adding module_variable_optional_attrs
to the list of active experiments.

**Comment by chgasparoto** [comment]
@jonasbenavides82 talvez se você atualizar a versão do Terraform e do provider da AWS funcione. Caso contrário, de acordo com a mensagem de error, você deve remover o `optional` da sua variável de objeto.

**Comment by yaratavares** [comment]
sei que é antigo, mas pode ser que alguém ainda encontre esse problema no futuro. Como o problema é o module, resolvi colocando o module_variable_optional_attrs no terraform:

```
terraform {
  ...
  
  experiments = [module_variable_optional_attrs]
}
 ```
